server/db/models/game.js

```diff
@@ -341,15 +341,21 @@ gameSchema.methods.dealCards = function() {
     .then(function(numCards){
         numCardsToDeal = numCards;
         cardsToDeal = deck.slice(0, numCardsToDeal)
-        self.update({deck: deck.slice(numCardsToDeal)});
-        return self.getPlayers()
+        self.deck = deck.slice(numCardsToDeal);
+        return self.save();
+    })
+    .then(function() {
+      self.getPlayers()
     })
     .then(function(players){
-        players.map(function(player){
+      return players.reduce(function(accumulator, player){
+        return accumulator.then(function(){
             var newHand = cardsToDeal.slice(0, 9-player.damage)
```

Comment:
If you splice instead of slice, the cardsToDeal array would be automatically mutated and you wouldn't have to do the slice on the line below (354)